src/core/src/op/reduce_logical_and.cpp

```diff
@@ -2,76 +2,61 @@
-                                           const bool keep_dims)
+namespace v1 {
+ReduceLogicalAnd::ReduceLogicalAnd(const Output<Node>& data, const Output<Node>& reduction_axes, const bool keep_dims)
     : LogicalReductionKeepDims(data, reduction_axes, keep_dims) {
     constructor_validate_and_infer_types();
 }
 
-shared_ptr<Node> op::v1::ReduceLogicalAnd::clone_with_new_inputs(const OutputVector& new_args) const {
+std::shared_ptr<Node> ReduceLogicalAnd::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v1_ReduceLogicalAnd_clone_with_new_inputs);
     check_new_args_count(this, new_args);
-    return make_shared<op::v1::ReduceLogicalAnd>(new_args.at(0), new_args.at(1), get_keep_dims());
+    return std::make_shared<ReduceLogicalAnd>(new_args.at(0), new_args.at(1), get_keep_dims());
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-namespace reduce_and {
-namespace {
-bool evaluate_reduce_logical_and(const HostTensorPtr& data,
-                                 const HostTensorPtr& out,
-                                 const AxisSet& reduction_axes,
-                                 bool keep_dims) {
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    out->set_shape(reduce(data->get_shape(), reduction_axes, keep_dims));
-    try {
-        ov::reference::reduce_logical_and(data->get_data_ptr<char>(),
-                                          out->get_data_ptr<char>(),
-                                          data->get_shape(),
-                                          reduction_axes);
-        return true;
-    } catch (const ngraph_error& e) {
-        OPENVINO_WARN << e.what();
-        return false;
-    }
-    OPENVINO_SUPPRESS_DEPRECATED_END
-}
-}  // namespace
-}  // namespace reduce_and
-
-bool op::v1::ReduceLogicalAnd::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+bool ReduceLogicalAnd::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v1_ReduceLogicalAnd_evaluate);
     OPENVINO_SUPPRESS_DEPRECATED_START
-    OPENVINO_ASSERT(validate_host_tensor_vector(inputs, 2));
-    OPENVINO_ASSERT(validate_host_tensor_vector(outputs, 1));
+    OPENVINO_ASSERT(inputs.size() == 2);
+    OPENVINO_ASSERT(outputs.size() == 1);
 
-    const auto& data = inputs[0];
-    const auto& axes = inputs[1];
-    const auto& out = outputs[0];
-    if (data->get_element_type() != element::boolean || !axes->get_element_type().is_integral_number()) {
-        return false;
-    }
-    const auto reduction_axes =
-        get_normalized_axes_from_tensor(axes, data->get_partial_shape().rank(), get_friendly_name());
-    OPENVINO_SUPPRESS_DEPRECATED_END
-    return reduce_and::evaluate_reduce_logical_and(data, out, reduction_axes, get_keep_dims());
+    const auto reduction_axes = get_normalized_axes_from_tensor(this, inputs[1], inputs[0].get_shape().size());
+    outputs[0].set_shape(ov::util::reduce(inputs[0].get_shape(), reduction_axes, get_keep_dims()));
```

Comment:
May it corrupt output shape in case of unsupported type? Previous implementation applied `set_shape` after type validation

Reply:
It will resize, but if evaluate return false anything in output cannot be treated as valid data or tensor. And usually fail in evaluate stop the whole  data processing.

I've seen these two approaches in evaluate, change shape before validation or after, in both cases there were no issues but check after type validation is part of template implementation which increase binary size.

Is also remove outside template implementation to reduce size.

Reply:
I propose to use has_evaluate at the begining of all the evaluates to solve it.

Reply:
It is used usually to check if evaluate can be called. Also `has_evaluate` assumes there are inputs set and evaluate will not work when operator has been created by default ctor.